src/Cli/dotnet/Program.cs

```diff
@@ -223,14 +223,14 @@ internal static int ProcessArgs(string[] args, ITelemetry telemetryClient = null
                 PerformanceLogEventSource.Log.ExtensibleCommandResolverStart();
                 var resolvedCommand = CommandFactoryUsingResolver.Create(
                         "dotnet-" + parseResult.ValueForArgument<string>(Parser.DotnetSubCommand),
-                        parseResult.UnmatchedTokens,
```

Comment:
Any idea why "parseResult.UnmatchedTokens" is not the same "as the rest of the result without the first command". Should it be a change in system commandline? parseResult.UnmatchedTokens should be used in this case

Reply:
When parsing `dotnet watch run`, `run` gets matched to the command so it doesn't get included in parseResult.UnmatchedTokens.

Reply:
So it skipped `watch` and find "dotnet run" could still make sense? Hmmm, that's a bit too smart. @jonsequitur what do you think? Is UnmatchedTokens simply not the right API call in this situation or we should dial down the smartness.